### DISCUSSION TITLE
Fluent Bit v2.0 is coming!

### DISCUSSION THREAD

**Discussion by edsiper** [fluent/fluent-bit#5993] [Announcements]
Dear Fluent Bit Community, 

We are planning the most significant release (ever) for the mid-end of October: __Fluent Bit v2.0__.

Fluent Bit is the new standard agent on most Cloud providers with huge community adoption. Continuing our journey to provide the fastest and low resource solution we are proud to announce some of the upcoming new features:

- Logs, Metrics & Traces: Full compatibility with Prometheus and OpenTelemetry
- Golang input plugins support
- TLS support for input/source plugins
- TAP: ability to create sessions and debug traffic live via HTTP custom HTTP endpoints
- WASM: create input and filter plugins in your preferred language ;) 
- Ability to run input/source plugins in a separate thread
- New integrations (output plugins)
- Many more bug fixes and as usual.. even more, performance improvements

Stay tuned to the updates but also feel free to post any questions you might have. We plan to do a code freeze around the end of September so you will be able to start trying out this new version before the official release!!

cheers!

note: If you plan to attend KubeCon NorthAmerica, let us know what's your t-shirt size ;)

**Comment by noniu** [comment]
Please support reload configuration files and do not need restart server.

**Reply by edsiper** [reply]
still in our to-do list :/

**Comment by ryn9** [comment]
With some of the architectural changes occuring in the core - will you be supporting more advanced message routing?
I had started a discussion here, https://github.com/fluent/fluent-bit/discussions/5802, but did not appear to gain any traction

**Reply by edsiper** [reply]
looks interesting, for 2.0 will not make it due to timings. let's wait for more comments from the community on #5802 to understand how we can make it happen

**Comment by sashaaKr** [comment]
@edsiper any estimation regarding when it will be released?

**Reply by edsiper** [reply]
from the description:

`We are planning the most significant release (ever) for the mid-end of October: Fluent Bit v2.0.`

**Comment by erdody** [comment]
@edsiper is this still on track for the end of the month? Also, anywhere I can read about not backwards compatible changes? Thanks!

**Comment by 666MrFang** [comment]
Ask a question, how to use Golang to create a new input plugin, is there any example? I currently only see examples of using Golang to create output plugins in fluent/fluent-bit-go

**Comment by hafkaM** [comment]
Is the kafka connector (kafka input) part of the docker image in version 2.1.4?
It tells me that it does not know this plugin. 
Or is it preparing for the next release?
Thanks M.

**Reply by patrick-stephens** [reply]
Not until https://github.com/fluent/fluent-bit/pull/5204 or something equivalent is merged.
You can review whether it is enabled by looking at the setting here for the relevant tag: https://github.com/fluent/fluent-bit/blob/v2.1.4/CMakeLists.txt#L172
